lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+import 'icons.dart';
+import 'message_list.dart';
+import 'store.dart';
+import 'theme.dart';
+
+void showNewDmSheet(BuildContext context) {
+  final store = PerAccountStoreWidget.of(context);
+  showModalBottomSheet<dynamic>(
+    context: context,
+    isScrollControlled: true,
+    builder: (BuildContext context) => SafeArea(
+      child: PerAccountStoreWidget(
+        accountId: store.accountId,
+        child: NewDmPicker(pageContext: context)))); }
+
+class NewDmPicker extends StatefulWidget {
+  const NewDmPicker({
+    super.key,
+    required this.pageContext});
+
+  final BuildContext pageContext;
+
+  @override
+  State<NewDmPicker> createState() => _NewDmPickerState();
+}
+
+class _NewDmPickerState extends State<NewDmPicker> with PerAccountStoreAwareStateMixin<NewDmPicker> {
+  late TextEditingController searchController;
+  late ScrollController scrollController;
+  final Set<int> selectedUserIds = {};
+  List<User> filteredUsers = [];
+
+  @override
+  void initState() {
+    super.initState();
+    searchController = TextEditingController()..addListener(_handleSearchUpdate);
+    scrollController = ScrollController();
+  }
+
+  @override
+  void dispose() {
+    searchController.dispose();
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void onNewStore() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  void _handleSearchUpdate() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  // function to sort users based on recency of DM's
+  void _updateFilteredUsers(PerAccountStore store) {
+    setState(() {
+      filteredUsers = store.users.values
+        .where((user) =>
+          user.userId != store.selfUserId &&
+          user.fullName.toLowerCase().contains(searchController.text.toLowerCase()))
+        .toList()
+        ..sort((a, b) {
+          final recentDms = store.recentDmConversationsView;
+          final aLatestMessageId = recentDms.latestMessagesByRecipient[a.userId];
+          final bLatestMessageId = recentDms.latestMessagesByRecipient[b.userId];
+
+          if (aLatestMessageId != null && bLatestMessageId != null) {
+            return bLatestMessageId.compareTo(aLatestMessageId);
+          }
+          if (aLatestMessageId != null) return -1;
+          if (bLatestMessageId != null) return 1;
+          return 0;
+        });
+    });
+  }
+
+  // Scroll to the search field when the user selects a user
+  void scrollToSearchField() {
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      if (scrollController.hasClients) {
+        scrollController.animateTo(
+          scrollController.position.maxScrollExtent,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.easeOut);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height * 0.95,
```

Comment:
It doesn't seem right to size it this way. If the goal is to avoid the top of the bottom sheet overlapping with the device's top inset, see `useSafeArea` on `showModalBottomSheet`.